README.md

```diff
@@ -50,6 +50,26 @@ highest precedence, these are:
 Providers can supply configuration to the worker via whatever means makes sense.
 For example, an EC2 or GCP provider would read configuration from the instance's userData.
 
+### Provider Configuration
+
+Providers that interact with the worker-manager service can get configuration from that service.
+That configuration formally has the form:
+
+```yaml
+worker:
+  config:
+    workerConfigValue: ...
+  files:
+    - ...
+```
+
+Where all fields are optional.  The contents of `worker.config` are merged into the worker configuration.
+Files are handled as described below.
+
+For backward compatibility, `worker` can also be spelled `genericWorker`; the two are treated interchangeably.
+Also for backward compatibility, configuration may be specified as a simple object with configuration properties at the top level.
+Both of these forms are deprecated, and support will be removed in a future version.
```

Comment:
Will worker pool definitions still be explicit about which worker implementation the config is intended for? (i.e. can an observer infer which worker implementation a worker pool executes by reading the worker pool definition in worker manager?)

Is there a means to provide configuration to applications running on a worker that worker-manager doesn't know about?

Reply:
> Will worker pool definitions still be explicit about which worker implementation the config is intended for? (i.e. can an observer infer which worker implementation a worker pool executes by reading the worker pool definition in worker manager?)

Only by inference from the names of the config parameters.  I think that would be better associated with the image in the configuration, rather than the config.  But I suppose you could also include a `workerImplementation: ..` key in the `workerConfig` if you'd like.

> Is there a means to provide configuration to applications running on a worker that worker-manager doesn't know about?

The idea is that these would be other top-level keys in `workerConfiguration`, aside from `worker`.

Reply:
>  But I suppose you could also include a workerImplementation: .. key in the workerConfig if you'd like.

We could even make that a feature of worker-runner, if you'd like -- if `workerImplementation` is present in `workerConfig`, then it could assert that the value matches the value in its config, and fail with an error reported to W-M if not.

In another PR :)